agent/rpc/peering/service.go

```diff
@@ -427,6 +431,19 @@ type BidirectionalStream interface {
 
 // StreamResources handles incoming streaming connections.
 func (s *Service) StreamResources(stream pbpeering.PeeringService_StreamResourcesServer) error {
+	if s.Backend == nil {
+		s.logger.Error("cannot establish stream without a backend", "error", errPeeringBackendNil)
+		return errPeeringBackendNil
+	}
+
+	if !s.Backend.IsLeader() {
+		// we are not the leader so we will hang up on the dialer
+
+		// TODO(peering): in the future we want to indicate the address of the leader server as a message to the dialer (best effort, non blocking)
+		s.logger.Error("cannot establish a peering stream on a follower node;")
+		return io.EOF
```

Comment:
Can we return something like `grpcstatus.Error` instead of a raw `io.EOF`? I find the EOF misleading since the stream hasn't even been established yet and it sort of pretends that it has been terminated.

Reply:
I was trying to find a "spec" like thing and at some point I saw `EOF` as a good way to indicate that the stream is closed.  I added a `grpcstatus.Error` w a failed precondition code. Wdyt?